oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/writer/DefaultIndexWriter.java

```diff
@@ -80,10 +83,22 @@ public DefaultIndexWriter(LuceneIndexDefinition definition, NodeBuilder definiti
 
     @Override
     public void updateDocument(String path, Iterable<? extends IndexableField> doc) throws IOException {
+        Iterator<? extends IndexableField> f = doc.iterator();
```

Comment:
The change to this method is causing 7 tests to fail.
```
[ERROR]   DefaultIndexReaderFactoryTest.indexDir:90 expected:<1> but was:<0>
[ERROR]   DefaultIndexReaderFactoryTest.indexDirWithBlobStore:128 expected:<1> but was:<0>
[ERROR]   DefaultIndexReaderFactoryTest.multipleReaders:174 expected:<2> but was:<1>
[ERROR]   MultiplexingIndexWriterTest.deleteIncludingMount:240 expected:<1> but was:<0>
[ERROR]   MultiplexingIndexWriterTest.deletes:209 expected:<2> but was:<0>
[ERROR]   MultiplexingIndexWriterTest.writesInDefaultMount:117 expected:<1> but was:<0>
[ERROR]   MultiplexingIndexWriterTest.writesInDefaultMountBlobStore:165 expected:<1> but was:<0>
```
These tests expect event the document only containing `:path` field to be added to index. I am not sure if that completely makes sense and what is the use case.
In addition to checking `containsOnlyPath` if i also check `definition.getPropertyRegex() != null` in lines 90 and 97 (i.e. before returning in case of reindex and before deleting the document in the other case), the tests pass.

Reply:
> These tests expect even the document only containing :path field to be added to index. I am not sure if that completely makes sense and what is the use case.

I don't know... However if feels a bit risky to change the behavior. What if we keep the current behavior (even if I agree it is weird), and open a new ticket to improve this area? So that documents that only contain :path are not indexed. You can then immediately add that part of your patch there.

Or, as you write, add the check definition.getPropertyRegex() != null in lines 90 and 97. That's also an option. (Even thought, decoupling the two aspects is a bit cleaner I think).